text/0093-nft-standard.md

```diff
@@ -0,0 +1,260 @@
+
+- Tracking ownership and metadata of NFTs. 
+- Transferring NFTs safely between accounts. 
+- Managing per-token approvals as well as operator-level approvals.
+- (Optionally) exposing metadata (name, symbol, token URI) so that applications can easily display NFT details.
+
+This proposal also addresses the backend indexing and tracking of NFT events (minting, transfer, approvals) for full visibility into asset histories.
+
+## Core Casper NFT Interface
+
+[core-casper-nft-interface]: #core-casper-nft-interface
+
+The core interface is designed to address the basic functionality required to manage non-fungible tokens. It uses Casper’s types (e.g., AccountHash, U256) and aims to capture the semantics of ERC-721, adjusted for Casper’s execution model.
+
+```rust
+#![no_std]
+
+use casper_contract::contract_api::{runtime, storage};
+use casper_types::{Key, URef, U256, AccountHash, ContractHash};
+
+/// Enum defining events that should be emitted by NFT operations
+#[derive(Debug)]
+pub enum CEP95Event {
+    /// Emitted when an NFT is transferred, minted (from=None), or burned (to=None)
+    Transfer { from: Option<AccountHash>, to: Option<AccountHash>, token_id: U256 },
+
+    /// Emitted when a specific NFT is approved to an account
+    Approval { owner: AccountHash, approved: Option<AccountHash>, token_id: U256 },
+
+    /// Emitted when an operator is approved/disapproved for all NFTs of an owner
+    ApprovalForAll { owner: AccountHash, operator: AccountHash, approved: bool },
+}
+
+/// Casper-compatible NFT interface
+pub trait CEP95 {
+    /// Returns the number of NFTs owned by a given account
+    ///
+    /// @param owner - The account to query.
+    /// @return U256 - The number of NFTs owned.
+    fn balance_of(&self, owner: AccountHash) -> U256;
+
+    /// Returns the owner of a specific NFT.
+    ///
+    /// @param token_id - The ID of the NFT.
+    /// @return Option<AccountHash> - The owner if it exists, else None.
+    fn owner_of(&self, token_id: U256) -> Option<AccountHash>;
+
+    /// Transfers the ownership of an NFT and performs a recipient check.
+    ///
+    /// @param from - The current owner of the NFT.
+    /// @param to - The new owner.
+    /// @param token_id - The NFT ID.
+    /// @param data - Optional payload to pass to a receiving contract.
+    fn safe_transfer_from(&mut self, from: AccountHash, to: AccountHash, token_id: U256, data: Option<Vec<u8>>);
+
+    /// Transfers the ownership of an NFT without checking the recipient contract.
+    ///
+    /// @param from - The current owner of the NFT.
+    /// @param to - The new owner.
+    /// @param token_id - The NFT ID.
+    fn transfer_from(&mut self, from: AccountHash, to: AccountHash, token_id: U256);
+
+    /// Approves another account to transfer a specific NFT.
+    ///
+    /// @param to - The account that will be granted approval.
+    /// @param token_id - The NFT ID.
+    fn approve(&mut self, to: Option<AccountHash>, token_id: U256);
```

Comment:
I propose to change `to` to `spender` since it was the term used in CEP-47 and CEP-78, and also to match the event parameter